src/extension.ts

```diff
@@ -30,43 +31,61 @@ export function activate(context: vscode.ExtensionContext) {
 
   const selectionDidChangeListener = vscode.window.onDidChangeTextEditorSelection(
     (e: vscode.TextEditorSelectionChangeEvent) => {
+      const TIME_TO_IGNORE_NAVIGATION_AFTER_MOVE_COMMAND = 500
       const timeSinceMoveToEdit = new Date().getTime() - lastMoveToEditTime
+
       if (timeSinceMoveToEdit > TIME_TO_IGNORE_NAVIGATION_AFTER_MOVE_COMMAND) {
         if (currentStepsBack > 0) {
           if (getConfig().logDebug) {
             console.log(
               `Resetting step back history. Time since move to edit command: ${timeSinceMoveToEdit}`,
             )
           }
+
+          if (getConfig().topStackWhenMove)
+            moveEditTopStackByIndex(editList.length - currentStepsBack)
+
           currentStepsBack = 0
         }
       }
     },
   )
 
+  const onConfigChangeListener = vscode.workspace.onDidChangeConfiguration((e) => {
+    if (e.affectsConfiguration('navigateEditHistory')) {
+      reloadConfig()
+    }
+  })
+
   const documentChangeListener = vscode.workspace.onDidChangeTextDocument(
     (e: vscode.TextDocumentChangeEvent) => {
-      const filepath = e.document.uri.path
-
       // actions such as autoformatting can fire loads of changes at the same time. Maybe we should ignore big chunks of changes?
       // TODO: perhaps there could be a smarter way to find autoformatting actions? Like, many changes that are not next to each other?
       // if (e.contentChanges.length > 30) {
       // return
       // }
-
       if (e.contentChanges.length === 0) {
         return
       }
 
-      // we only use the last content change, because often that seems to be the relevant one:
-      const lastContentChange = e.contentChanges[e.contentChanges.length - 1]
-      handleContentChange(lastContentChange, filepath)
+      // iterate over all changes, nessisary to keep old edits line numbers up to date
+      e.contentChanges.forEach((change) => addEdit(change.text, change.range, e.document))
```

Comment:
I've previously had some problems with this when triggering stuff like auto formatting that could result in a bunch of edits. But now that I tested it wasn't that bad. So lets try a release with this, and see how it works out.